django/db/backends/base/creation.py

```diff
@@ -268,6 +272,13 @@ def destroy_test_db(self, old_database_name=None, verbosity=1, keepdb=False, num
             settings.DATABASES[self.connection.alias]["NAME"] = old_database_name
             self.connection.settings_dict["NAME"] = old_database_name
 
+    def _restore_initial_data_migrations(self):
+        # Load initial data migrations, if any.
+        if hasattr(connections[self.connection.alias], "_test_serialized_contents"):
+            connections[self.connection.alias].creation.deserialize_db_from_string(
+                connections[self.connection.alias]._test_serialized_contents
+            )
```

Comment:
Any feedback about my previous comment @timgraham? I will continue to work on this issue in the next days, if you think that's the right way.
I will test it manually with different configurations.